modAion/src/org/aion/zero/core/IDifficultyCalculator.java

```diff
@@ -38,5 +39,5 @@
  */
 @FunctionalInterface
 public interface IDifficultyCalculator {
-    BigInteger calculateDifficulty(AbstractBlockHeader current, AbstractBlockHeader dependency);
+    BigInteger calculateDifficulty(A0BlockHeader current, A0BlockHeader dependency);
```

Comment:
Good point, I have had another look, here IChainCfg needs to be also refactored in order to separate the PoW specific definitions and AbstractBlockHeader needs a getHeader method in order to be able to be used also with non Aion cases, which should be the correct case.

Reply:
Because difficulty is a PoW-specific term so the dificulty calculation should use the PoW-specific implementation of the block header, which is `A0BlockHeader`. And as @centrys-bsafta  said, we should continue refactoring `AbstractBlockHeader` and move PoW-specific properties like `solution` to `A0BlockHeader` so it can be reused in the future.

Reply:
Agreed re: refactors to `AbstractBlockHeader` and `IChainCfg`.

It makes sense that IDifficultyCalculator works with classes that are PoW-specific.  I think A0BlockHeader is Aion-specific though.  I also noticed that there's also IPowBlockHeader.  Would it make more sense for IDifficultyCalculator#calculateDifficulty to operate on those instead?